src/main/java/com/spartronics4915/lib/lidar/LidarProcessor.java

```diff
@@ -287,25 +291,20 @@ private void processLidarScan(LidarScan scan)
             } 
             else
             {
-                Pose2d estimate = mEncoderStateMap.getFieldToVehicle(scan.getTimestamp());
-                // need to invoke getFieldToLidar, not getFieldToVehicle
+                // XXX: this needs validation
+                // XXX: need to invoke getFieldToLidar, not getFieldToVehicle
                 Transform xform = mICP.doICP(getCulledPoints(scan), 
-                                new Transform(estimate).inverse(),  // ie: LidarToField
-                                mReferenceModel);
-                pose  = xform.inverse().toPose2d();
-
-                Pose2d distFrom1SecBeforescan = mLidarStateMap.getFieldToVehicle(scan.getTimestamp() - 1).inverse().transformBy(pose);
-                vel = new Twist2d(
-                    distFrom1SecBeforescan.getTranslation().x(),
-                    distFrom1SecBeforescan.getTranslation().y(),
-                    distFrom1SecBeforescan.getRotation().getRadians());
+                                new Transform(lastPose).inverse(),  // ie: LidarToField
+                                mReferenceModel); // mReferenceMode in field coords
+                Twist2d fwdK = Pose2d.log(xform.inverse().toPose2d());
+                poseEstimate = lastPose.transformBy(Pose2d.exp(fwdK));
```

Comment:
Nitpick: I think we could reduce how much we operate on what gets integrated into pose here, to reduce accumulation of floating point errors. Maybe:
```java
Pose2d xform = mICP.doICP(getCulledPoints(scan),
                                 new Transform(lastPose).inverse(),
                                 mReferenceModel).inverse().toPose2d();
poseEstimate = lastPose.transformBy(xform);
velMeasured = Pose2d.log(xform);
```

Reply:
i believe the exp is actually needed, since it moves along an arg, whereas xform is a linear transformation.